orca-peering/src/main/kotlin/com/netflix/spinnaker/orca/peering/PeeringAgent.kt

```diff
@@ -156,43 +160,49 @@ class PeeringAgent(
+      var hadFailures = false
+      var orchestrationsDeleted = 0
+      var pipelinesDeleted = 0
+
+      try {
+        orchestrationsDeleted = destDB.deleteExecutions(ExecutionType.ORCHESTRATION, orchestrationIdsToDelete)
+        peeringMetrics.incrementNumDeleted(ExecutionType.ORCHESTRATION, orchestrationsDeleted)
+      } catch (e: Exception) {
+        log.error("Failed to delete some orchestrations", e)
+        peeringMetrics.incrementNumErrors(ExecutionType.ORCHESTRATION)
+        hadFailures = true
+      }
 
-    log.debug("Found ${deletedExecutionIds.size} (orchestrations: ${orchestrationIdsToDelete.size} pipelines: ${pipelineIdsToDelete.size} deleted candidates after cursor: $deletedExecutionCursor")
-    var hadFailures = false
-    var orchestrationsDeleted = 0
-    var pipelinesDeleted = 0
+      try {
+        pipelinesDeleted = destDB.deleteExecutions(ExecutionType.PIPELINE, pipelineIdsToDelete)
+        peeringMetrics.incrementNumDeleted(ExecutionType.PIPELINE, pipelinesDeleted)
+      } catch (e: Exception) {
+        log.error("Failed to delete some pipelines", e)
+        peeringMetrics.incrementNumErrors(ExecutionType.PIPELINE)
+        hadFailures = true
+      }
 
-    try {
-      orchestrationsDeleted = destDB.deleteExecutions(ExecutionType.ORCHESTRATION, orchestrationIdsToDelete)
-      peeringMetrics.incrementNumDeleted(ExecutionType.ORCHESTRATION, orchestrationsDeleted)
-    } catch (e: Exception) {
-      log.error("Failed to delete some orchestrations", e)
-      peeringMetrics.incrementNumErrors(ExecutionType.ORCHESTRATION)
-      hadFailures = true
-    }
+      if (!hadFailures) {
+        deletedExecutionCursor = (deletedExecutionIds.maxBy { it.id })
+          ?.id
+          ?: deletedExecutionCursor
 
-    try {
-      pipelinesDeleted = destDB.deleteExecutions(ExecutionType.PIPELINE, pipelineIdsToDelete)
-      peeringMetrics.incrementNumDeleted(ExecutionType.PIPELINE, pipelinesDeleted)
+        // It is likely that some executions were deleted during "general" peering (e.g. in doMigrate), but most will be
+        // deleted here so it's OK for the actual delete counts to not match the "requested" count
+        log.debug("Deleted orchestrations: $orchestrationsDeleted (of ${orchestrationIdsToDelete.size} requested), pipelines: $pipelinesDeleted (of ${pipelineIdsToDelete.size} requested), new cursor: $deletedExecutionCursor")
+      } else {
+        log.error("Failed to delete some executions, not updating the cursor location to retry next time")
+      }
     } catch (e: Exception) {
-      log.error("Failed to delete some pipelines", e)
+      log.error("Failed to delete some executions", e)
+      peeringMetrics.incrementNumErrors(ExecutionType.ORCHESTRATION)
       peeringMetrics.incrementNumErrors(ExecutionType.PIPELINE)
```

Comment:
ah hmmm... it wouldn't be terribad to pass an execution type to `peerDeletedExecutions` and call it twice, right? Then the metrics wouldn't have to lie 😅

Reply:
it's a bit sucky since there are both pipelines and orchestrations in that `deleted_executions` table...